### PR TITLE
Replace Gruntfile.js with simpler Gruntfile.coffee

### PR DESCRIPTION
**Pull request by firstdoit**
The major change is that Gruntfile.js was replaced by a leaner Gruntfile.coffee. In detail:

- Use `grunt` default task for dev, instead of grunt serve
- Use `grunt prod` instead of `grunt serve:dist` for dist preview
- Use grunt-contrib-connect `base` instead of arbitrary mountFolder function
- Use declarative tasks object

### CODE REVIEW

**Comment by newtriks** [comment]
Anyone wanna chime in with some preferences on this kindly created pull request :monkey_face:

File: test/test-creation.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ describe('react-webpack generator', function() {
       react.run({}, function() {
         assert.fileContent([
             ['package.json', /react-hot-loader/],
-            ['Gruntfile.js', /hot:\s*true/],
+            ['Gruntfile.coffee', /hot:\s*true/],
```
Comment (ColCh) [inline test/test-creation.js]: With coffee, `hot: true` may be replaced to `hot: yes` or `hot: no`. This check is becoming *unsafe*.

**Comment by andrewliebchen** [comment]
I'm a designer, and I use this generator to stand up prototypes quite a bit. The syntax of CoffeeScript might be a lot simpler, but it is something else to have to learn. I'm okay with CoffeeScript, but most examples for Grunt are in JS, so it's just one step harder to get things done.

I do like the flattening of the `src/` directory, though. :smile:

**Comment by skfd** [comment]
This should be an option. I'm customising `Gruntfile.js` after generation and I will not be able to do it comfortably if it's suddenly some other language like TypeScript, Dart or Coffeescript.

**Pull request closed without merging**